controllers/domachine_controller.go

```diff
@@ -283,6 +283,14 @@ func (r *DOMachineReconciler) reconcile(ctx context.Context, machineScope *scope
 			return reconcile.Result{}, err
 		}
 		r.Recorder.Eventf(domachine, corev1.EventTypeNormal, "InstanceCreated", "Created new droplet instance - %s", droplet.Name)
+	} else {
+		// do we want to retag control plane and worker capi droplets?
```

Comment:
The control plane droplets definitely need to be retagged. Since the DOCluster.LB depends on it. But, should the workers also be retagged?

Reply:
Have we tested and observed the migration bug effecting the worker droplets too?

Reply:
Yes. The migration affects all domachines irrespective of whether it is control plane or worker as we observed during a recent incident. From the code in this controller, you can see that we build the tags when creating worker droplets and one of the tags contains the cluster uid.

Reply:
But, worker droplets frequently scale up and down and when new ones come up, it uses the correct uid in the tag.

Reply:
Ok, then it's probably good to not worry about reconciling them too. But maybe we should add a regression test for both control plane and worker domachines to ensure the bug is actually fixed with your change and confirms the worker assumption that we're going with here.

Reply:
I think I slightly disagree. :) I think we need to fix this completely. And not rely on autoscaler scaling a cluster up and down to fix a migrated cluster.

Reply:
I agree with that. The scaling up and down of the cluster should not be the method of reconciliation here.

Also feel free to add tests, but if it proves to be too tricky. We can always follow up.

Reply:
Fully agree, we need to fix it for both. When migrated the old UID is gone and therefore incorrect and won't ever come back.
Keeping it would just be confusing and if used by a user would cause trouble and unexpected behavior.